#include <iostream>
#include <string_view>

std::string_view input = R"({{{{{{{<>},<<!>},<,>},{{<oou<o{!>,<i!!!!!>!!ouu}i>},<a!!!>!i!!!>!!<!!,!oa''!!!>!{>}},{{{{<a!!!>!!!>!!"!!o">}},{{{{{<!!i!!!>},<!!!>!>},<>}}},{}},<!>},<iu<!>!!o!!!!ee!!!!io!!!>!>!>},<'>},{{{{<!>,<ei'{!!}!>},<!><!!!>>},{<"a!>!>,'>,{{{<!>},<!>,<!e>}}}}},<!>},<!!!><!!i!!e"!>},<!>},<,!>,<!!!>!>a,>},{{{<>}},{{<!,u<!}!>},<!>},<i!>,<e>}}}}},{{<!>!o{a"i!u{}!>,<{!!!>,<!!!>!a>,{}},{{},<!>,<!>},<!!a,e{a!!!>e>},{<!eo,!!aa>}}}},{{{{{}}},{}},{{{{},<e!!!>>},{}},{{{<<!!ii!!!>i'!"a!!!>!>>}},{{{},<}{e}<!>},<>},{<!!o<!!!u!!"},!!,{!>},<!>!!!!!!!>!!!>},<>}},{<"e!>},<,o!!"!>e!>}"!>,<}a>}},{{<!!a'!!!!uo""!!'{!!!!'>}}},{{<,!!{ai{,ou!>},<a!>,<!!>,<>},{{{<'o!!!><!!!>u,!!!>!>},<!!!!!>,<>}},{{{{<>}},{{{<!!"aei!!!!!>"!!o"!!!!!>aioo,>},<!>!!!>,<!<,!!!>{i!!!>},<!<!!!!!!u'{!!,e"e>},{<{!>},<}!>,<!!!!o!!!>!!i}!!{!!!>>}}},{{{{<u!}<,o{!!!!,<{!!'>},<a>},{<<!>},<e!>,<'}u,!e}!>},<!!!>">}}},{{{<>}},{{}}}}}}},{{{<ae{,}<>},{{<!>},<}!eu}"},!a,a>}},{{{{<e!!ue"'<i!>,<!!ai!>},<"">},{}}},<,{>}},{{{}},{<!>!>},<!!!{!"'!uu!!!>},<{>}},{{},{{<<o!>,,i!!!!u!!'!'{"ei{">},{}}},{{{<!>,<}<!{,!>},<i!>,!!e}">}},{{<a!u!!!>!eu"!{!ua{>}}}}},{{{{},{}}},{{<!!!>!<au!!!!,!i"!!!>},<o}i,!!'{<'!>,<<>},{{{{},{<aiu,!>,<!!>}},{{{<!e"!a"i>},{{<!!!>u,a{!!!>,<eu}!!<}!>>}}}}}},{{<}!>u!>},<eooo!!a!!'!!!o{u!>},<>}}},{},{{{}}}},{{{{},{{{<{"o!!o!><'!!!>>},{{<!!e!!,}ia!>},<'!!,'<e!>},<u!>!>,<!!!>!>,<!!!>>},{<!u!!,u<!>},<!!!>u!!!>!>,<u>}}}},{{{{<u!!},,u{'!!!!!!,!!'>},{<<',!oi!>},<!>,<!oi!>!>},<e>}},{{}}}}}},{{{},{{{{}},{{},{<!>},<!!!>!!!>"<i!!>}}},{{{<!!!>!>},<u!,i!!!!!>!>},<aa!!{!!!ie"!>!">,<!!!>!!ee>},{{<!><!!!><{!!!>'e!!<!>,<!>,<'!{>}}},{{<}u,!,!!!>!>},<!!!>!"i{!!!>,>}},{{<"{!!!>ioua!!!>u!ouu!>a!!u>}}},{{}},{{{{},{<e>},{{<!>!!!!!!!>,<"!!!>ea<aa!>},<'!!!>ue!'>}}},{{{{<!>,<ei!!!>},<o!!ei!"!!<!>},<u}>}},{{{{{},{<i'!>},<!>,!>},<"oa<>}}},{{<au"}!>,<!e"ieau>}}},{<",o<!>!!!>e,!>,<uo!>},<eae!>},<",!!!i>,{<e!!<!!<uiu>}},{<!,<}ae}!>,<ue}i}!!!>{>}},{<!!!}'!,o}!'iu!>},<!!!>!'o>}},{},{{{<'>},{<>}}}},{{{{{},<>},{<{!oo"!>},<<"e!>,<!>"{o!}!!!>},<a!>>}},{<!!!>,<,"e!!i{ii!!!!a}u>,<!!!eooio!!!>!>,<'o{!>!>,<!>,<,>},{{<'!!!>!i!!!>"!>},<ieo!!!!!>,<!>'!>,<u!>},<}>}}},{<!>},<"}!>},<!!o>,<uo!!{<'!{a!>,<>},{{{<!>},<'u!>},<>},<!!}e!>'a!!!>!o!!}">},{{<i'{{}ea{"e{!!!e!!!}!>>}}}},{{<<!!!!!e!!{"'!>},<!!!!'u!!a!"!>,<!>},<<oo!!ie!!!>>,<u!!!>!!a{>},{{{{<>}},{}},{<!!!!a!>,<"<<<!!i"!!'"}!!!au'>}},{{{},{}}}}},{{{<{}a!u!>},<!>},<ei!!!>{!!'>},<!!!!!>!!!!!>oo''<>},{{<!>},<!!!!!!,!>,<a!>},<'>},{{<u{!>!!!>!>},<a!,ae!!!!!>{>}}}},{},{{{{{},{<u!>,<!>oio!}!>,a!!'u!>},<'!!>}},{<u!>,<!>!!!>},<}!>!!!>!!!!<{!!io!>!!!>},<!!io>},{{<i{!>!!!>>}}},{<i>,<i!!ou'!>},<!!!}ea!>!!"'"{!!,>},{{<,"!!a!>},<!>,<!>},<!>},<!!!>>},<{!!ou'!!!,a>}},{{<!!o',e!!,a>},{}},{{{<!>},<!!!>o{u"!!!!!!!!!!>}},{<!!!!!>>}}}}},{{{<e!!},au!!ee!a!!{!!u>},<o!!!{!"!!"i>},{{},{<!>,!>,<>,{{<!",e<o'<iao,i!>,<!!!>>,{<i{!!!>!!!>!>!!!>!>},<!}a}!ioe>}},<}!!e>}}}}},{},{{},{{},{<uii",i!!!o'">}}},{{<}!>},<a!!,!>,<!!!>!!!>a!!"!!'>},{<{'!>},<oau!!!>i!>,<!!"<{!>'!>},<'!!>,<>},{<!a{o!>},<!{!a>,{<u,"!!!>,>}}}}},{{{{{{<""!!!>!>,<'!o!!!>>},{}}},{<'!"o!!!>}!>},<>,{<'>}},{{<,a!i"!}i!!!>"a"!!!!{u!!,i>},{<o!>},<!!!>},<{io!>e!>,<!!!>,<>,{{<o'!!ioa!!ia"""'>}}}}},{{<!!!>!<au!!,!!"e!!!>!a,eeia,>},{<>}}},{{{<>}},{{<!>},<}>},{<o'a>}},{{<<!<!>},<{!>!ia!>},<{o!>}!!!>},<>}}},{{{<!>'o"a}euu!!!>ua>},{{<"!>},<{!>a!>},<i}!>,<eo!>},<>},{<}!!e!a!!>}}},{{<!>},<!!!>,<!!!<i"!!<!>,<!>,<!>,<!!}!>,<<oo>,<!>,<"!!!>,<!>},<aiio!!,!>}!u!>},<u!>,<u!!"a{>}},{{<>,{<!oi>}},{{{<}i!'",!>!,!!e<!!!>!u>}},{}},{<ie!>>,{<!>,<"'!'e!!!>!>},<!>},<u"!>!!<{>,{<e'!>,<!>'!>},<'!!!!a{{!!!>!>e}a}!>},<!>,<>}}}}},{{{<!!!>!>,<!<''>},{<o!!!>!,!>},<!>u''<>}}}},{{{{{<!>e"a>}}},{<!>!!}!>,<'o!!!>,<!!!>}""o{u!!!>>}},{{{<!!au"!!}!!!!!>},<{!!e,o!>!ou>},{{{<!!>}},{}},{{<"'!'{>},<!>},<o>}},{{{<!>,<'a,!eo,!!>},<""!>,<!>},<!,'!!!><!i!>,<e{>},{{<!!a!>!!!>ii!>},<!>,<u,!!i'!!i'!!<!!{!!!ea}!!!!!!>,{}}},{{},{{<!>,<!>},<!}<,{ua,<!>},<e!!e<>},{<!!!>uo"!!!>,}e!>u>}}}},{{{{<}!!}!>}'!!<{!a!>,<a!{!,>}}}},{{{<{"<!ua{!>},<"!ii"'}!!!>!!}<!>,<<>,<,!!!!!!!!!>"{!!"eu,!!!>,<o"a!!i!>,<e>},{}},{},{{},{{{<u!>u<!!<!>},<!>,<!!!>}">},<""e}<,!>!o!!!>}!!o>},{<{>}}},{{{{{<!a}uuee>}},{{{<,o!>},<u!!!>!>,<!}!!!>a}ee>},{{<{,!!,!>},<!!!>a>},<!><'<i!>!!u<a!>{i>}},<!!'oe!>},<<u!!!!>}},{{<aa!>},<!o!>},<"{o<>},{<!>},<!!!u"!>,<>}},{<}oio>,{{<!<{}<"e!!!>'!>},<o!"i,!i!!!!u>},{<!>,a!>!>},<!!!>,<}>}}}},{{<{,ei!e{'{!{">,{<i!!!!!>,o!",a}!>,<!!ia!!>}},{{<",'!>{!"iu!!!>!>},<!>},<!!!>>,{<"a!!!>!>!>},<}{!,!!!!ou}{>}},<!!ie<!!!!uuu!u,!>u}!!!>>}},{{<"!>,<!>,<!o{!>},<!!!>a!!a,!>,<u""a!><">,{<!!o!!<'!!!>!>,<!!!>!>},<!!ae!>},<!!!>},<!!!><'>}}}}}}}},{{{{{<!}!o{!e!!<<o!!o!!,!>},<e>},<i!>},<"o!>!>},<'!!!>,<a!>},<e>},{{{{<au,'!!!>!>,<!!a>},{<"!}!i!!!>},<o!>},<o}'!!!>!>},<{>}}}}},{{{<i!>"!!ua!>},<"!!!>!!!>{'ie{'}>},<!!!>,!!!!!>},<u!>!!u"{'o>},{{<u!!!!!>!>},<!!a!a!!{{'<!!!!!>,<!e>,{<!!!!!>o'ui!!!><{>}},{{},{<!>,<<u!>{o!!}"!>},<{}e">,{}}}}},{{{<ao'"!!!!!>!>},<!!!'"u>},{{{{<!!!>,<>}}},{}}},{{<}!>,<'!>,<!!iu!u!!!>,<!}ae}!i!!!,!>!>a>},{<!!!>!>,<"<e}!>,<<!!i>},{<!!!>!>,<!io>,<{!!"eo>}},{{},{{<!!,!!!>o!io!>},<!!!>'<!!!'!!i>},<'i,!>iea!!!!!>"!!!'>}},{{{{<e!>,<}!!"!>,<<a!e'}<>}},{<!>},<,'!>,<u>}},{{<o!!!>,}!!'!>},<!>},<{>},{<aa'u!!}!>>}},{{{{<!>'!!"i!!!>!"o!!!>!!!>ii'>}}},{<u!!!>},<a!>,<u},!>!!!!!>!>,<a<!{!>,<{<u>}}}},{{{<!i!!!>a}">}},{{<u{!>a!>!>,<{!!!!i!!!>,<!!!>,{>},{<u,e!!,<i!!!au!>},<'}}!>,<!!}o>}}}},{{{{{{},{<!>},<'{!!>}},{<a!a!!!>!!{!>o!!e!!"'!!!>},<"o!!<>}},{{<!!}au"!>>,<!eo!>,!><!!!!'i<,!!!>,<}!!<{!io}>},<'"!>,!!!>!>},<e,eei!!}!!<!>,<o!!!>>},{{<e<aoaua,!!!>aa!!!>eau,'>,<i!ua"u}>}}}},{},{},{{{<o'!'o!e!!!!'ea}>,{}},{{<,{<e!!!!}}<!>},<>},{}},{{{}},<!!!>!>!!"'!!u!a!!,a>}},{{},{<!!!>oo"!!!>,{>,<!>'<!>>},{<o,!!!!!'!!!>!!!>!!'{''!>{i{!>},<>}},{{{<!>,<"!!'i"u<!{o<i{i!}!>i>},{}}},{{{{<'<{<iaa!>},<!>},<{a!!!>>}},<o!!!>!!{!!o>},{{<!!!>},<i!>!!!!e!>}!!!>!!!!!>'">},<>},{{<'{!!}a">}}}}},{{{{{<<!>,<"!>},<u!>},<!!!>,<!i'{>}},{<,!!!>},<au!>},<!!ie,!!}!>,<>,<,!>},<!!!>,<a>},{}},{{<i"u,!{ae{,!<a'!>},<!>,<>},{{{{{<o"!>ua,!}'<<u!a!>,<u}!>>},<i!>},<i!!!>o"!>},<">},{{{},{}}}},{{<!!o!,,!a}>},{{<>},<!!"!>},<!{"}!>},<!!>}}},{},{{{{<!!a}!!e!<uu!}!!eu!!{ai{!>!'>},{<'!!!>!!!>},<!!!!!>"ioa!!!>}!!!>'>}},{}}}}}},{{<<!>,<!!!>!o'e{!!""!e',a!>!!!>},<>},{{},{<!i!!ui<""}!!!{oi,!>,<<{o!!u>}}},{{{}},{{<{>},<!!{!!!>,<!u!!,!>},<i!!ui!>o!>a,>},{<!>o!>e'!!!>},<!!<>,{{}}}}}},{{},{{{<o!!!><,!!"}!>},<!!!><u!>,<>},{{}},{<!}{!!!'a>,{<!!!!!!au,!>},<i!>,<!>,<<aaa!e>,<}!!!!!>e"{!!<,'!>!>!!!>"!ee{!>},<>}}},{{{<a!>,<!o},ai!>},<>}}},{{{{<!>,<e<!>},<,e!!!>}'!!'uei>},{<"{i>}},<!,!{!>},<"!>,<u!!!<!!!a!!!>eu!<{a>}}},{{{{<{"!oi>,{}},{{<}!>},<"au>,{<!!}{io<!a!!!>,<<ai!!{!eo>}},{}}},{{{<!!!>i!!!!}'e!!!>"}<!u!}a}>,{<<!'"ea!!!>!!!>>,<>}},{<!>},<au!!!>},<e!!o!!a,!>!o!>,<>,<!>,<}!>},<!>!!!>!!!>!>},<>}},{}},{{{{{},{}}},{{<!>',ua'!!a{{>},{<!,!!!>{>}},{{<}!>aai!!'"}!<!!!>!!<!!}uiue>},{{{<!,!>},<{!!e!>},<!e{!!}!!u<a"!>,<u"i!>,<>},<>}}}},{{{<o!>},<!!!>e!!{'"oi!>>}}},{{{<!!},!!!a!!!>!!i,o!>},<!!!>>},{{},{<!!ii,!!!>,<"!!!>}!>},<!!u>}}}}},{}},{{}}},{{{<!>o!e,!!>,{{},{}}},{{<!<>},<}!>},<u'u<ea!i'<o<!>oeee>}},{{{<i,{a<'!>i"e{!>},<o>}},{}},{{{},{<!>u'!!!o}a!>,<>}},{{{{<,oo!!!>o<"ao!!!>,<'!}!u!!,,{>}},{{<u!!u!!!><!!!>},<},>},<i"!!'e!>},<u!>},<!!!!!>},,!>{>},{{{{<}"""e!>ei"!"}!!e'!!>}},{<!!!>},<{<}}ieoai!",!>,<u>},{{{<!!,{!}!i!>},<'}!}u{,!}{e!!>,{<,}}!>!>!,i!>},<!{"o}u!!"i!!a}>}},{<a!>o,ua!!!>,<u!>},<"!!!!a!}a!!!>},<{!!!>!>,<>}}}}}}},{<!!!>!>},<i<!!!!!>},<e!>o}!!!>,<{!!<}u>,<!!!!!{!!o!ooe!!!!!>!!}}!>!>},<a!},!!!}'!!,i>}},{{<aa!!!>!>,<a'!!!><!>},<!!!>},<{!!!!!>'o!>,<>},{{{<,<!!!{<eaa!>'>},{}},<a{!>!!},!!!>'>}}}},{{{{{{<!{e!>>,{<!!!!!>!<!>},<!!!>,<!!!>,<>}},{{<!>},<!!!!!>a,!!}!"i{!!!>,<>}},{{<o!>,<!>}!!!!>}}},{{{{{},{<!>,<e!!!!!>}!!!>,<!!!>,<!!iei>}},{{<!!!>e!>!>i!!}!>,<>},{<i!!!!!>,<,!!ou!!<!!<e"{e<>}},{{}}},{{<!u<!!!><}u'!!!>},<,!!!e>}},{}},{{<!>,<a!!i>},<}u""!!!>io!!!>},<!!e!,!!>},{<!!!>{>,{}}},{{}},{{{<!eu!!!>o,!!e''i<!!!'!>},<!!!>>}},{<!!!>a!>,<ao!!!>,<>,<,!,}>},{<!!!>o'!>e>,{}}}},{{{{}}},{{{<u{,'">,{}}},{{{<o,!!!>!>>}},{<'o}'!>,<!i>,<!!a!>,<!!!!'{!!!>{a!!i'>}},{{},{{{<,!>},<!!!!!!,!>},<!>,<!>,<i>}},{{<!!!>u!!!>o!>ei,!}!!!!!!!>!!!>!>,<>},<'o!}'>}},{{{<i!>,<<>}},{{<!!{!>},<!>},<!!!>},<!!!>,<u!>},<!>!!!>ae!>,<>},<!!!>u!>,<<'<!"}!!!!!>{!>!!"}!>,<<>}},{{{<!',<!!ua!"'!!a!>},<,!o!!u!!!}'i!!>,<o!!o>},{{{},{<!!i!u">}},{}},{}},{{{}},{{<"u!e!!,!,!>},<i!>},<i'{!!!>,a!>},<"!>,<!>>},<!>u!>},<!!{!>!>,<!><}!!i!>,<i'a'!!,e>}},{{{{{{{},<}}},!,<e}!!!!}!i!>>}}}},{{{<a}e!>},<<!!!>>,{{<!!!!<,{e!>!!!>!!a",!>},<u{>}}},{<!!!>"<e<!!oo!!!>!>a!!,!>},<>}},{<"!!!>},<,!u!!!{!>},<,<,o'!>,<"!"{!>,<>,<!>!e"!!!!!>!!!!}"<<!!!!!>'o!>},<"!>},<}u>},{{{{<,!>,<a,e!!!>!<!!,>}}},{{<!io{>}}}},{{{}},{<!>,<!!!<!!eu'!!!>}!>},<,i{!>},<!!!>>,{{},{<!!}'!!!>i!>,<i!!!}!>!>!!i!{!>,<!!!!o!!!>},<!!!>>}}}}}}}}},{{{<>}},{{<>},{<!>,<oa'o!!e<,!>,<a!>,<>}}},{{<!!e!,i!<"!<<!>},<}!!!>!>>,<!!!}e!!!>,<o!!!>!!>},{<o'o!>!>},<e{!>!>!!!>},<!!!>!e!!!>!>!>o!!"!a>,{<!>,<!!!!!>>,{<'>}}}}},{{{<u!>!!,e!!>}}},{{},{{<!>},<{ia!!}e"<!>!>},<!!i"{!>,<!!e!">}},{<!"}'!>,!>},<!<!!'!>u!'!!i!!!"!<>,<!>!!ua}<!!<u"!!!!!!!>{}>}}},{{{{{{{},<>},{<!>!!!>!!a!>,<!!!><>},{{{<,,ua!>,<<"!i!{"!>},<oi>,{<>}},{<}}!!!>!>o!!!>!!a!!{i<ua,{a{>}},<<e!>,<!>},<i!!!>!!!>}!!'!!!!!>,<u<!!<!!!!}!!u>}}},{},{{{<!!}e!>,<>}}}},{{<!!!>!!!>,<ii!>,<!!!aaaa'!>},<!!,!<!>,<!>!>}>,<!!u!ui!<!!!>},<<<!>,<ia!>e!!u!!{<>},{{<uuia"}u!>,<o!>,<{!>},<!!!>!'!>},<'{,">,{<!>},<<,ue!!<!e!>,<!>},<!!!a>}}},{{{<!!!>"!!e!>,<!>,<{!>,<>},<!>},<'a"!!!,!>},<!!'>},<'"uo{e!!{!!!>i"!>!{!>},<,i{>}},{{{}},{{{<!!!>!>,<>,<iou!!!>!!!>!>,<!,u!!!>!>},<>}},{}},{{<!>},<!>},<!>i<!!o{u!!!>,<a!i!!!><>}}}},{{<,oa!>o"!>},<a!!a}>},{<'!>},<ii">,{<,u>}},{{<a!!!>,<o}!!!>,!<'>},{{<!!e!!,!ui!o,>}}}},{{<>}},{{{},{{<!>},<<!,!!!>!!!!!!!>},<e!<!>},<!>!!'>},<o">},{}},{{<"!i!!ii!!!>,<,!!!>'{,'u<>}},{{<<!{'"!>},<>},{<,!e!>!!!>i!>,<!>,<!!"!>},<!>!}!>!>},<oa{">}}}},{{{{<<>}},{{<!>!!u!!!!},!>},<!>'<!>,<!!{"i"'!!!>!>},<o>},<!>,<!o!!!>>}},{{{{<}{a!>},<uu}<!>},<!>},<,"ia!'!!,e'i>}},{{<!!>},{<!>},<{!>,<ii,e'a!>},<!!"!>!>,<">}}},{<!>,<{"!>!>,<!i!>>}},{{{{<ea!!}i'>}},{},{{{{<!!!>!!"a!>,<>,{}}},{<!>},<e'>}}}},{{<!!"'{{!!o,'u!,>,{<a!!{!>>}},{{{{{<}"!!!>!>u!!!>}}i!>,<}'a!{{>},<,!>!>!}<i!>,<!!!>},<a,>},{{{{<>}}},{<!>,<!!eao,!!!!'>,{<i!!'",!",a!au>}},{{<!!',!!oa!!!>a!,ei!!!>!>"!>>}}}},{<!!,!!!>!>},<!>!>}}""<!!,eo!>},<>,<a!!!>{!>},<!"<!'}>},{<o>}},{{<e}!>},<u!>!>},<!!"!!<'!!i!>},<!>,<!}!>{,!>},<>}}},{{},<!>!!!>'}eu!>,<!a!!!>!!!>!!!>!!!>"}!e>}},{{{{<!,!}!!!e<!!!>!>,<i!i>},{{<!!!>,<{!{i!>}!}a",oo!>,<!!!>,<>,{<,!e'>}},<!>},<!i!>},<!<<!!!>a!!!!'<'!!!>,<>}},{{<e!>},<!!!!}>}},{{<o!,{o!!!!,!e<i!>,<u'<i!!!!e>},<!u!>,<!>},<!!!>},<"i!!e!>},<iu}!>},<i>}}},{{{<!>{}"!!!>},<!!u<!>,<"!<a,{!!!>>},{{<!>},<>,{<a"'!>},<{"!}!!oo!!!>'!>,<u'!>,<'>}}}},{{{<!!!!u"ioa,!>},<!!!>!>},<!!!>},<!'!i!!u>},<{!>,<u!!<}!!!{auau,}i>},<a<!!ou!>,<!!!!i{!>,<<>},{{}}}}}},{{{{<!!u,'<!!!>},<<u!!!'!',!>,<"!>,<!!!!!>}>},<!>,<!!!!<,!!!!!!}!>{'!>},<ue!>,<!ao!!!!!>>},{{<!!}uo!>,<}i!{>}}},{{{{{},{}}},{{<a!>{>,<!>,!!!u!>>}}},{}},{{},{},{}}},{{{{{<a!>,<e,!!"!!!>},<!!{}}!!,io{!ioou>}},{{}},{}},{{},{{},{{{}},{{<<!!!>},<,'!!a!>},<!!!>'!!!>>,<!>',!>,<e,<,<uiaaa!>!!'!>},<>},{{{<!!!>'!!!>!>},<e!>ue!e"!!!>},<<uu!!!>"o>},{<!!!>oe}{'!!<>}},{<>}},{{},<'!>},<!!!>u>}},{<>}},{{{{},{{{<,a!>e!!!>!!!>}o!>},<e!,!!e}"!!!>,<>}},{{<}!!!!!!!>o<>}}}}}},{{{},{<{!!!>'!!u!!!>'>,{{<!o!>'!!!!e!"i>}}}},{{{<u>,<!!,o!!e!!o{o!>,<>}}},{{{{{{<!!!>}!>},<!!{!!!,>},<!!!!u!>o''"u>},{{{<!!!>!!a!>,<'{!>,<!>,<>}},<<!!""!!o!!iauu!>},<i!!!>ii!,!!e>}},{}},{{<!!!!!"!>},<{i}!>},<!!e{ea!!!>!"<e!!u>}},{{{{}},{<,!>i!!i,a!>},<"!>u!aoe>,{<,<!!>}}},<i!>!>,<'!ua!o!>,<!u!!>}},{{{<ea"!!!>!>},<!,,!!!>!!oie"!i>,<!!!>{!!!>'{{>}},{{{{<!>!!o!>i!!!>},<<e!!'!>,<e>}}},{},{{<!>},<!!}>},<uo!>>}},{{{<'!!!!!>o!!!!!>!>,<!>},<!!!>>,{<}!!!!!!u!!!>},<!!!>,<!{!!"!!!>>}},{<>,{<eii{!!'eua"i!!!>'>}}},{{{<e!>ee}i!a!!!>!!!>},<}<!!!>!!,!!!>!>!!i!>,<i>}},{{{{},{}}},{{{{<!!!>e{uo{u>,<!>!!!>,<{!>}!u>},<!!!>'>},<,!!u!>,<!!}aiua!!!>,<!!!>>}}},{{}}},{}}},{{<,{{>},{{<,!>},<i!!!>iu{}!>,<!!!>!!!>!!!>{!{a,>,{{<!!!>!aa>},<au"ua'e!!!><!>},<!u!!!>o!!'!,>}},{{},<"'ai!!!>'ia>}},{{{},{<u!>},<"!!!'!>!!!!}!>},<!!!>!!!>,<uo!>,<>}},<!>},<}'!><o{!>,<i{!>!!,>}},{{{<!!!>"}>},{<ie!>,<}}ieo!!o!>,<!!!>!!!>>}},{<"!>!e!!!>!>},<!!{,!>,<!!!!!>,!<>,{<>}},{<!>},<!!a}!!{"!>,<!>!<}<o!!!>{'>,<!'"o>}}},{{{{<!>,<!!!}!!}!>,<eo<}>},{{<e!!!>,<!e}!e{!!!>},<!>o!!<!!'!!!{{<!><>,{}}}}},{{<}'!!!!'!<o!>>}},{<,u!!{!!!!!>"!!e>,<!'<!"!>},<}"}<!!!>!!{o!!!>o!ie!!!>>}}}},{{<!!!>},<!!!>i!>e<!>!<!>,<!!!!{"uao"i!>,<!>},<,>,<u!!{{!!'o!>},<!!,}>},{{<}ei!>,<o{!!!>!}u!>},<!>},<!!o{!>,<iui!>},<>},{<!,oo''}!!!>},<!!!>i!>},<a!!!>u}u>}},{}},{{{{{{<"io!>},<!>,<"}!"!!!>}}>},{<>}},{}},{{{{<!>!>},<!>,<!!uo!!o!>,<!!eu!>o!{>},{<a<iu,!!aa!ea!!!!!!!>!u<i<,!!"!!!o!e}>}},{{<u!>,<e<>}}},{<a!>},<!>e'<!!!!!>!!!!'!a!>},<}i>,{{},{<<a!}!o"!>,<<u!>},<!>},<!!!>!!a"e>}}},{{<!a",i<>},{<!io!>,<>,<'ai!">}}}},{{<e!>a}}!"!!e!!o,e!>,<i{!!!>,<{<ua>},{<!<i"!!!>!!>}}},{<!!!!!!u>,<e!!!>,<<eu>},{{{<!!}{!>},<''!!!>!!{!!!><!!i!>},<a,!!!>},<'!!!!!>},<>},{}},{{<!!!><!!ou,!>uea>},{}}}}},{{}},{{{{{{<u>}},{{},{<}!!!>,<u>}},{{{<!>u!{io!>,<!!a!!!>{{>},{{<}!ea!>>}}},{<!!!!!>,}!!!!!>!!!},!>,<,!!{'!!!o''<>}}}},{{{{{},{{}}},{}}}},{{{}},{},{<'u!!!>!!!!!>},e}{io>,{}}},{{{{}},{<!!"}>}},{{},{{{{},{{}}}}},{{{},{{}},{<{!!!"!!!>!ea!!!<>,{}}}}}}}}},{{{{{<!>,<a!>!>!"}!!>}},<!!!>!!!!!!!"!"i,!>,<>},{<!!!!!!<!!!>i!!!>{e!>a'i{!!!><i>},{}},{}}},{{},{{{<!uo{{o!!!e!>},<!"eo>,{<!!>}},{<u,>,<,'aa{!!!>},<!!!!!!a!!'',i}'!>!>},<!!}ei>},{{<!>},<!>e>}}},{<{!!!>!>!!>}},{{<!!!>,<"{oo{!!!>,!>},<i!>>,{<i<"iu!>a'!!!!}!>,<o!!!>ii!!"!>>}},{{{<'oo"}!!!>},<ee'!!!>{e!!"!>,<o>},<iui'!!!>},<oa>},{<o<u!!o<!!!>!!}uu!>},<i>}},{{{}},<{<!>},<i!!!>u<u!>},<!>!>,<,{<,{>}},{{{{{}},<}o!{<}eoe!>'{a!>,<!>},<>},{{{<{!>,<!>!}!!!!}>,{<ue{!>,<!>},<!>},<!!!>o!!'!>,<!}!,>}},<"!>!>e!>},<"">},{<i}!!u!!!!!>,<!!!>!!a'<!>},<!>},<{"!!!>!!>,<!!!>!!{>},{{{}},<oeae!>,<!"!!!>'!>,!!!><!}!>},<!>,<u>}},{}},{{{{<"'"},'}!>},<"!,!!!>!>,<!!!!!>!>,<!!a>}},{}},{{<!"!!,!>},<!!{'<u!!!>{!><!>},<!>ie!>,<!!!>},<>}}}}}},{{{{{<!!!>au>,{<!!!>!>{"!>,<!>},<{{a'a"{,o!!!>!!<>}},{},{}},{{<>}}}},{{{<!>!!!>!!!>>,<!ie}!!!>>},{{{<"ei!!!>,u>},{<<<!!!>,<!!!>ia!>},<!!!!!>"">}}},{{{{<a!>,<u{'!i!>},<!o>},{{{<eu!!!>},<>},<!!!!"!!!>!!!>i,!!!>o"ao!>,<e'!!i>}}},{<'!!o">}},{{{<>},<!>a>},{{<!!"!>,<u!>,<!>,<}>,<!>,<ou{!'{!,}!>,<>}}},{<'<e<{''i!>},<!!!>,<u!>,<!>,<<<"!!<>,{<!!aoa!!!>!!'!>{!>},<'i!!!>>}}}},{{{{<'e}u"}a!!!>!!!>,<!!!>},<}!!!>>},{<}e!!!!}a{i>}},<i<a">},{{<<!>,!>!i!!!>},<!!eu>,{{<!!!>{}!!"!>},<!>!!!>!!!>"!!!!>}}}},{{<}!!>},{<i,!>},<!u,!!u>}}},{{{<!>,<!!!>!!!>!!!>,<}!!>},<!>},<uo<!>},<!>},<i!u!!!>!}"<!!!!!>o,>}}},{},{{{{},<!>,<,eeu!>,<,"u'"!>},<">},{{{<!!!!a!a!>,<!>,<!>},<!>,<!!e!>,<>}},{}},{{{{{}},{{<ea!!e!>oo!>o!!!!!!!>{!!u"'">}}},{<o,}!!e!>},<a!!!!u!!'{io>}}}},{{{<>,<'a,}!!!!!<'!!!>},<e!!{!>,{!!"!>!>},<!!!}>},{{<a>},{<!!}"!!!>},<!>},<o<o!>},<i!>},<!!!>,<!!!>"!!<{!!o>}}},{{<!>e}{!{!>,<!>,<{<!!!}!!!>u}e!>e!ua>},{{<"}'!>,<u!!!>{,!>}}{!!!!!>,<!!{}>}}}},{{{},{{},{<'!!!>}!>},<oa!!o"!>},<!>,<!'!!!>,<u!>,<o!>!!<!>>}}},{{{},<!>!>,<,<!!o}{>}},{{{<>}},{<!"ou!,,i!>,<a'!>"'!><{>},{<,!!,!!!!!>"!!!>!>},<"!!!>!>>,{<!>,<!>},<,!>},<!!!>}!!"!>,<!>},<<<!>u'<!!!>>}}},{{{<}{!>,<<"!!!>!!{!>,<>},{{},{<"!>},<{e}>}},{<"a!>},<e!>},<'{!!!!!>i!>,<!>>,{<!!!>,{,!!!>!!a!>},<!>},<>}}}}}}}})";

int main() {
    int level = 0;
    int score = 0;
    int total_garbage = 0;
    std::size_t loc = 0;
    while(loc < input.size()) {
        switch(input[loc++]) {
            case '{':
                ++level;
                break;
            case '}':
                score += level--;
                break;
            case '<':
                while(input[loc++] != '>') {
                    if(input[loc-1] == '!') ++loc;
                    else total_garbage++;
                }
                break;
        }
    }
    std::cout << "Part 1: " << score << '\n';
    std::cout << "Part 2: " << total_garbage << '\n';
}
